single_tech_samples/databricks/sample4_ci_cd/README.md

```diff
@@ -1 +1,256 @@
+
+**It is the scaffolding of Azure databricks project.**
+
+To make it easy extendable, the notebooks and python code only contain super simple logic, and the unit testing is implemented by pytest and [nutter](https://github.com/microsoft/nutter)
+
+This template focuses on automating provisioning, CI/CD pipeline, and various approaches of databricks implementation.
+
+### 1.1. Scope
+
+The following list captures the scope of this template:
+
+1. Sample code
+    1. Notebook with pyspark API.
+    2. Notebook with spark sql.
+    3. Multi-NotBooks (main notebook plus reusable notebooks)
+    4. notebooks with python module
+    5. Non-Notebook, purely python
+2. Unit testing
+    1. pytest for python module
+    2. [nutter](https://github.com/microsoft/nutter) for notebooks
+3. Provision Azure Databricks environments of development, testing and production via provision pipepline.
+4. CI/CD pipelien to build and run testing automatically.
+
+Details about [how to use this sample](#3-how-to-use-this-sample) can be found in the later sections of this document.
+
+### 1.2. Architecture
+
+The below diagram illustrates the deployment process flow followed in this template:
+
+  ![architecture](images/architecture.png "architecture")
+
+### 1.3. Technologies used
+
+The following technologies are used to build this template:
+
+- [Azure Databricks](https://azure.microsoft.com/en-au/free/databricks/)
+- [Azure Resource Manager](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/overview)
+- [nutter](https://github.com/microsoft/nutter)
+
+## 2. How to use this template
+
+This section holds the information about usage instructions of this template.
+
+### 2.1. Prerequisites
+
+The following are the prerequisites for deploying this template :
+
+1. [Github account](https://github.com/)
+2. [Azure Account](https://azure.microsoft.com/en-au/free/search/?&ef_id=Cj0KCQiAr8bwBRD4ARIsAHa4YyLdFKh7JC0jhbxhwPeNa8tmnhXciOHcYsgPfNB7DEFFGpNLTjdTPbwaAh8bEALw_wcB:G:s&OCID=AID2000051_SEM_O2ShDlJP&MarinID=O2ShDlJP_332092752199_azure%20account_e_c__63148277493_aud-390212648371:kwd-295861291340&lnkd=Google_Azure_Brand&dclid=CKjVuKOP7uYCFVapaAoddSkKcA)
+   - *Permissions needed*:  The ability to create and deploy to an Azure [resource group](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/overview), a [service principal](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals), and grant the [collaborator role](https://docs.microsoft.com/en-us/azure/role-based-access-control/overview) to the service principal over the resource group.
+
+   - Active subscription with the following [resource providers](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers) enabled:
+     - Microsoft.Databricks
+
+### 2.2. Setup and deployment
```

Comment:
Just a thought.. Should we move this section to each samples as the deployment steps may be slightly different for each sample. We can just cover high-level concepts in the parent Readme.

Reply:
yes I remove it and add link to the readme of provisioning